docs/strategies/hedging.md

```diff
@@ -471,6 +471,45 @@ Some insights about the hedging callback behavior in relation to hedging context
 > [!NOTE]
 > The hedging strategy ensures that both `ActionGenerator` and `OnHedging` are never executed concurrently. Any modifications to the contexts are thread-safe in these callbacks.
 
+### Sequence diagram about contexts and callbacks
+
+For the sake of conciseness the `Caller` is not depicted on the diagram.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant P as Pipeline
+    participant H as Hedging
+    participant AG as ActionGenerator
+    participant UC as UserCallback
+    participant HUC as HedgedUserCallback
+    participant OH as OnHedging
+
+    P->>H: Calls ExecuteCore <br/>with Primary Context
+    H-->>H: Deep clones <br/>Primary Context
+
+    H->>+UC: Invokes <br/>with Primary Context
```

Comment:
note: in this call the Primary = Action context, they are filled but these are the same instances.

But I am thinking, is this OK? 
Shall we have the same behavior even for primary action? Just clone the primary context and use it for primary callback too. wdyt?

Reply:
I think that approach would be cleaner as the original request won't be treated in a special way.

Reply:
True, I have prepared the PR with the changes: #1754 

Can you also align the diagram with it?

Reply:
Done